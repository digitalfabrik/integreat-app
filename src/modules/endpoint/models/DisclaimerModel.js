class DisclaimerModel {
  constructor ({id, title, content}) {
    this._id = id
    this._title = title
    this._content = content
  }

  get id () {
    return this._id
  }

  get title () {
    return this._title
  }

  get content () {
    return this._content
  }
}

export default DisclaimerModel
