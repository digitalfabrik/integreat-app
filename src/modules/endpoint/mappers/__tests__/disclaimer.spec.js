import disclaimerMapper from '../disclaimer'
import DisclaimerModel from '../../models/DisclaimerModel'

jest.unmock('../disclaimer')

describe('disclaimer', () => {
  const pageJson = {
    id: 1689,
    permalink: {
      url_page: 'feedback-kontakt-und-moegliches-engagement'
    },
    title: 'Feedback, Kontakt und m√∂gliches Engagement',
    type: 'disclaimer',
    status: 'publish',
    modified_gmt: '2017-06-12 12:27:57',
    excerpt: 'Excerpt',
    content: '<span>Content</span>',
    parent: 0
  }

  it('should throw if there are multiple disclaimers', () => {
    expect(() => disclaimerMapper([pageJson, pageJson])).toThrowErrorMatchingSnapshot()
  })

  it('should throw if there is no disclaimer', () => {
    expect(() => disclaimerMapper([])).toThrowErrorMatchingSnapshot()
  })

  it('should throw if the disclaimer is not published', () => {
    const unpublishedPage = {...pageJson, status: 'no published'}
    expect(() => disclaimerMapper([unpublishedPage])).toThrowErrorMatchingSnapshot()
  })

  it('should map fetched data to models', () => {
    const disclaimerModel = disclaimerMapper([pageJson])
    expect(disclaimerModel).toEqual(new DisclaimerModel({
      id: pageJson.id,
      title: pageJson.title,
      content: pageJson.content
    }))
  })
})
