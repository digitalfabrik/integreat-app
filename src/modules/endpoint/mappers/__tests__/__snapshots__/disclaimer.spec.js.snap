// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`disclaimer should throw if the disclaimer is not published 1`] = `"There must be exactly one disclaimer!"`;

exports[`disclaimer should throw if there are multiple disclaimers 1`] = `"There must be exactly one disclaimer!"`;

exports[`disclaimer should throw if there is no disclaimer 1`] = `"disclaimer:notAvailable"`;
